Note are not saving in custom app

**Describe the bug**
Notes are not saved in the tunisie app home page but saved on other pages of that zim file.


https://github.com/kiwix/kiwix-android/assets/34593983/6bbdc7a2-26a4-4ce3-943b-1bd5d840ad36



**Expected behavior**
It should save the notes on the home page as well.

**Steps to reproduce the behavior:**
* Download tunisie app.
* open application.
* try to save notes.
* you will see the error.

**Environment**
- Version of Kiwix Android: 3.7.0
- Device: Redmi Note 9
- OS version : Android 12

**Logs**
```
W/System.err: java.io.FileNotFoundException: /storage/emulated/0/Android/data/org.kiwix.kiwixcustomcustomexample/files/Kiwix/Notes/wikipedia_fr_tunisie-app_maxi_2023-01/Portail:Tunisie/Index thématique.txt: open failed: ENOENT (No such file or directory)
W/System.err:     at libcore.io.IoBridge.open(IoBridge.java:574)
W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:236)
W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
W/System.err:     at kotlin.io.FilesKt__FileReadWriteKt.writeBytes(FileReadWrite.kt:108)
W/System.err:     at kotlin.io.FilesKt__FileReadWriteKt.writeText(FileReadWrite.kt:134)
W/System.err:     at kotlin.io.FilesKt__FileReadWriteKt.writeText$default(FileReadWrite.kt:134)
W/System.err:     at org.kiwix.kiwixmobile.core.main.AddNoteDialog.saveNote(AddNoteDialog.kt:332)
W/System.err:     at org.kiwix.kiwixmobile.core.main.AddNoteDialog.onViewCreated$lambda-4(AddNoteDialog.kt:268)
W/System.err:     at org.kiwix.kiwixmobile.core.main.AddNoteDialog.$r8$lambda$44O8fA-doWM8bqseKW7UjVxIiQQ(Unknown Source:0)
W/System.err:     at org.kiwix.kiwixmobile.core.main.AddNoteDialog$$ExternalSyntheticLambda1.onMenuItemClick(Unknown Source:2)
W/System.err:     at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
W/System.err:     at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:779)
W/System.err:     at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
W/System.err:     at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
W/System.err:     at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
W/System.err:     at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:975)
W/System.err:     at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:623)
W/System.err:     at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
W/System.err:     at android.view.View.performClick(View.java:7751)
W/System.err:     at android.view.View.performClickInternal(View.java:7724)
W/System.err:     at android.view.View.access$3700(View.java:858)
W/System.err:     at android.view.View$PerformClick.run(View.java:29336)
W/System.err:     at android.os.Handler.handleCallback(Handler.java:938)
W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
W/System.err:     at android.os.Looper.loopOnce(Looper.java:210)
W/System.err:     at android.os.Looper.loop(Looper.java:299)
W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:8280)
W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1073)
W/System.err: Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
W/System.err:     at libcore.io.Linux.open(Native Method)
W/System.err:     at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
W/System.err:     at libcore.io.BlockGuardOs.open(BlockGuardOs.java:274)
W/System.err:     at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
W/System.err:     at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8153)
W/System.err:     at libcore.io.IoBridge.open(IoBridge.java:560)
W/System.err: 	... 29 more
```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt
@@ -179,7 +179,7 @@ class AddNoteDialog : DialogFragment() {
    * For null input or on being unable to find required text, returns null
     */
   private fun getTextAfterLastSlashWithoutExtension(path: String): String =
-    path.substringAfterLast('/', "").substringBeforeLast('.', "")
+    path.substringAfterLast('/', "").substringBeforeLast('.')
 
   // Override onBackPressed() to respond to user pressing 'Back' button on navigation bar
   override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

